front/ui/storybook/stories/ui-core/ComboBoxCustomSuggestions.stories.tsx

```diff
@@ -0,0 +1,454 @@
+      const baseMatchesTrigram = trigramUpper.includes(baseUpper);
+      const baseMatches = baseMatchesName || baseMatchesTrigram;
+      if (!baseMatches) return null;
+
+      const chList = suggestion.chList.map((ch) => {
+        const codeUpper = ch.code.toUpperCase();
+        return {
+          ...ch,
+          isCandidate: codeUpper.startsWith(lastTokenUpper),
+          isBestSuggestion: false,
+        };
+      });
+
+      // we apply scores depending on how suggestion matches base and last token
+      const trigramExact = trigramUpper === baseUpper;
+      const trigramPrefix = !trigramExact && trigramUpper.startsWith(baseUpper);
+      const trigramContains = !trigramExact && !trigramPrefix && trigramUpper.includes(baseUpper);
+
+      const namePrefix = nameNorm.startsWith(baseNorm);
+      const nameContains = !namePrefix && nameNorm.includes(baseNorm);
+
+      let baseScore = 99;
+      if (trigramExact) baseScore = 0;
+      else if (trigramPrefix) baseScore = 1;
+      else if (trigramContains) baseScore = 2;
+      else if (namePrefix) baseScore = 3;
+      else if (nameContains) baseScore = 4;
+
+      const chExact = suggestion.chList.some((ch) => ch.code.toUpperCase() === lastTokenUpper);
+      const chScore = chExact ? 0 : 1;
+
+      return { suggestion: { ...suggestion, chList }, baseScore, chScore };
+    })
+    .filter((x) => x !== null);
+
+  const preliminary = scoredPreliminary
+    .sort(
+      (a, b) =>
+        a.baseScore - b.baseScore ||
+        a.chScore - b.chScore ||
+        a.suggestion.name.localeCompare(b.suggestion.name, 'fr')
+    )
+    .map((x) => x.suggestion);
+
+  let candidateCount = 0;
+  let bestSuggestionIndex = -1;
+  let bestChIndex = -1;
+
+  preliminary.forEach((s, sIndex) => {
+    s.chList.forEach((ch, chIndex) => {
+      if (ch.isCandidate) {
+        candidateCount += 1;
+        if (candidateCount === 1) {
+          bestSuggestionIndex = sIndex;
+          bestChIndex = chIndex;
+        }
+      }
+    });
+  });
+
+  // If there is not exactly one candidate, nothing is highlighted
+  if (candidateCount !== 1) {
+    return preliminary;
+  }
+
+  // If there is exactly one candidate, highlight it
+  return preliminary.map((s, sIndex) => ({
+    ...s,
+    chList: s.chList.map((ch, chIndex) => ({
+      ...ch,
+      isBestSuggestion: sIndex === bestSuggestionIndex && chIndex === bestChIndex,
+    })),
+  }));
+}
+
+const ComboBoxStory = (props: { small?: boolean; disabled?: boolean; readOnly?: boolean }) => {
+  const maxVisibleSuggestions = 8;
+  const [value, setValue] = useState<Suggestion>();
+  const [filteredSuggestions, setFilteredSuggestions] = useState<Suggestion[]>(() =>
+    filterAndMarkSuggestions(ALL_SUGGESTIONS, '')
+  );
+  const [pickedCh, setPickedCh] = useState<string>();
+  const pendingPickedChRef = useRef<string | undefined>(undefined);
+
+  const valueRef = useRef<Suggestion | undefined>(value);
```

Comment:
You don't need to specify the type here
```suggestion
  const valueRef = useRef(value);
```